lib/auth/authentication_backend.rb

```diff
@@ -0,0 +1,110 @@
+module AuthenticationBackend
+  # Abstracted Exceptions
+  # Aws::CognitoIdentityProvider::Errors::NotAuthorizedException,
+  # Aws::CognitoIdentityProvider::Errors::UserNotFoundException,
+  # Aws::CognitoIdentityProvider::Errors::InvalidParameterException,
+  # Aws::CognitoIdentityProvider::Errors::CodeMismatchException
+  # Aws::CognitoIdentityProvider::Errors::ServiceError
+  # Aws::CognitoIdentityProvider::Errors::AliasExistsException,
+  # Aws::CognitoIdentityProvider::Errors::UsernameExistsException => e
+  # Aws::CognitoIdentityProvider::Errors::ResourceNotFoundException
+
+  class NotAuthorizedException < StandardError; end
+  class UserGroupNotFoundException < StandardError; end
+  class AuthenticationBackendException < StandardError; end
+
+  class AuthenticationResponse
+    attr_reader :params, :id_token, :access_token
+
+    def initialize(params: nil, cognito_response: nil)
```

Comment:
This suggest the arguments are optional (nullable), which doesn't look like it
```suggestion
    def initialize(params, cognito_response)
```

Reply:
I has pushed up a fix for that... You're right they should not be nil.